### PR TITLE
fix: loading insecure secrets from config

### PR DESCRIPTION
**Pull request by iain-anderson**
**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-sdk-c/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [x] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

File: src/c/secrets-insecure.c [hunk 1]
```diff
@@ -34,12 +34,12 @@ static iot_data_t *insecure_parse_config (iot_data_t *config)
     if (strncmp (key, SEC_PREFIX, SEC_PREFIXLEN) == 0)
     {
       size_t l = strlen (key);
-      if (strcmp (key + l - 5, "/path") == 0)
+      if (strcmp (key + l - 11, "/SecretName") == 0)
```
Comment (edward-scott) [inline src/c/secrets-insecure.c:37]: I can guess at what the magic number 11 means but would prefer it was calculated somehow even if only in a comment. Also there seems to a define of SEC_PATH that was maybe related once?

File: src/c/secrets-insecure.c [hunk 2]
```diff
@@ -34,12 +34,12 @@ static iot_data_t *insecure_parse_config (iot_data_t *config)
...
       {
         iot_data_map_iter_t iter2;
-        char *prefix = malloc (l + sizeof ("Secrets"));
+        char *prefix = malloc (l + sizeof ("SecretData"));
         strcpy (prefix, key);
-        strcpy (prefix + l -4, "Secrets/");
+        strcpy (prefix + l - 10, "SecretData/");
```
Comment (edward-scott) [inline src/c/secrets-insecure.c:42]: Ditto on the magic number.

**Review by edward-scott — CHANGES_REQUESTED** [review]
See inline comments.

**Comment by iain-anderson** [comment]
Re-used the old macro for the path element and removed the magic

**Pull request merged**